src/blr/cli/tagbam.py

```diff
@@ -1,58 +1,100 @@
 """
-Transfers barcode sequence information from the input file alignment name to SAM tags in output file.
+Transfers SAM tags from query headers to SAM tags. Currently tags in header must follow SAM tag format, e.g.
+BC:Z:<SEQUENCE>.
 """
 
 import pysam
 import logging
+import re
 from tqdm import tqdm
+from collections import Counter
+
+from blr.utils import print_stats
 
 logger = logging.getLogger(__name__)
 
+ALLOWED_SAM_TAG_TYPES = "ABfHiZ"  # From SAM format specs https://samtools.github.io/hts-specs/SAMtags.pdf
+
```

Comment:
Here we allow for any SAM tag type, we are however only using "Z" or strings at the moment. I would assume that only including this would speed up the regex search.